File: 2016/d03_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'd03'

class TestD03 < Minitest::Test
  def setup
    @example = <<~EXAMPLE
      101 301 501
      102 302 502
      103 303 503
      201 401 601
      202 402 602
      203 403 603
    EXAMPLE
  end

  def test_part1
    assert_equal 3, D03.new(@example).part1
  end

  def test_part2
    assert_equal 6, D03.new(@example).part2
  end
end

File: 2018/test/aoc2018/d03_test.clj
(ns aoc2018.d03-test
  (:require [clojure.test :refer [deftest is]]
            [aoc2018.d03 :as d]))

(def example "#1 @ 1,3: 4x4\n#2 @ 3,1: 4x4\n#3 @ 5,5: 2x2")

(deftest part1-test
  (is (= 4 (d/part1 example))))

(deftest part2-test
  (is (= 3 (d/part2 example))))
